src/components/UserSelector.tsx

```diff
@@ -2,15 +1,77 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classnames from 'classnames';
+import { getUsers } from '../api/users';
+
+type Props = {
+  selectedUser: User | null;
+  onSelectUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectUser,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [error, setError] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  const handleOutsideClick = useCallback((e: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(e.target as Node)
+    ) {
+      setIsDropdownOpen(false);
+    }
+  }, []);
+
+  const handleUserSelect = useCallback(
+    (user: User) => {
+      onSelectUser(user);
+      setIsDropdownOpen(false);
+    },
+    [onSelectUser],
+  );
```

Comment:
The method `handleUserSelect` does not follow the naming conventions for methods as mentioned in the checklist. Consider renaming it to `onUserSelect` or another name that follows the conventions.